torch/autograd/__init__.py

```diff
@@ -140,6 +140,23 @@ def grad(outputs, inputs, grad_outputs=None, retain_graph=None, create_graph=Fal
         inputs)
 
 
+# This function applies in case of gradient checkpointing for memory
```

Comment:
does this explanation sound reasonable @ezyang ?

Reply:
I think it can be done more clearly but it is good enough for now. One final suggestion might be to link to the PR with all the gory details.

Reply:
added the link to PR for discussion.